src/pages/index.js

```diff
@@ -91,18 +199,67 @@ profileAddButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-// addCardCloseButton.addEventListener("click", () => {
-//   newCardPopup.close();
-// });
-// previewImageCloseModal.addEventListener("click", () => {
-//   imageCardPopup.close();
-// });
-
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  const cardElement = card.getView();
-  return cardElement;
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+
+    handleLikeClick
+  );
+  return card.getView();
+}
+function updateProfilePicture() {
+  updateProfilePicture.setEventListeners();
 }
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+profilePictureValidator.enableValidation();
+//get user info from server and set it locally on the page
+api
+  .loadUserInfo()
+  .then((info) => {
+    userInfo.setUserInfo(info);
+    userInfo.setUserAvatar(info);
+  })
+  .catch((err) => {
+    console.error(err);
+    alert(`${err}. Failed to get user info.`);
+  });
+
+// get cards from server and render them on the page
+let section;
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards.reverse(),
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+    alert(`${err}. Failed to add card.`);
+  });
+
+function handleLikeClick(card) {
+  api
+    .likeCard(card.id, card.isLiked)
+    .then((res) => {
+      card.setIsLiked();
```

Comment:
Needs correction: `setIsLiked` method should have an argument - the isLiked value, you can get it from the response `res`